roles/matrix-dimension/defaults/main.yml

```diff
@@ -39,7 +39,7 @@ matrix_dimension_integrations_rest_url: "https://{{ matrix_server_fqn_dimension
 matrix_dimension_integrations_widgets_urls: ["https://{{ matrix_server_fqn_dimension }}/widgets"]
 matrix_dimension_integrations_jitsi_widget_url: "https://{{ matrix_server_fqn_dimension }}/widgets/jitsi"
 
-matrix_dimension_homeserver_federationUrl: "http://matrix-synapse:8048"
+matrix_dimension_homeserver_federationUrl: "http://matrix-synapse:{{matrix_synapse_container_default_federation_port|string}}"
```

Comment:
Leaving this default variable in `roles/matrix-dimension/defaults/main.yml` is probably OK. This way, we won't have to refer to `matrix-synapse` variables inside the `matrix-dimension` role. We usually try to avoid using variables across roles like this and prefer to hook roles together using the `group_vars/matrix_servers` file.

It may even make sense to define `matrix_dimension_homeserver_federationUrl` as empty in the `matrix-dimension` role and use `roles/matrix-dimension/tasks/validate_config.yml` to validate that it's defined.

You can redefine `matrix_dimension_homeserver_federationUrl` via `group_vars/matrix_servers`:

```
matrix_dimension_homeserver_federationUrl: "http://matrix-synapse:{{matrix_synapse_container_default_federation_port|string}}"
```

Reply:
Ah nice ! Yes leaving `matrix_dimension_homeserver_federationUrl` in `matrix-dimension` role as empty and define it in `group_vars/matrix_servers` is better.